packages/squiggle-lang/src/rescript/Utility/Errors.res

```diff
@@ -0,0 +1,31 @@
+type property = {fnName: string, propertyName: string}
+
+type rec error =
+  | NotSorted(property)
+  | IsEmpty(property)
+  | NotFinite(property, float)
+  | DifferentLengths({fnName: string, p1Name: string, p2Name: string, p1Length: int, p2Length: int})
+  | Multiple(array<error>)
```

Comment:
I don't think `Errors.error` is a good name for a type of errors that only relates to pointset, if i'm understanding correctly that that's what's going on. `Errors.poinSetError` is much better.

Reply:
Also why not put this file in the `Distributions/PointSet` subdirectory?

Reply:
A few things:
- It's possible to call XYShape outside of PointSet
- IsEmpty() and NotFinite() and similar seem like pretty generic errors; first for arrays, second for floats. 

I could see maybe something like we have a bunch of basic errors, then XYShape has its own versions of some of these, like,

```
xyShapeErrors = IsEmpty(ArrayErrors.IsEmpty(p)) | DifferentLengths(ArrayErrors.DifferentLengths(a,b))
```
Not sure though.

Reply:
I have a preference for more specific errors, just because they create more honest types (for instance, you don't need to check a case that can never occur). I prefer your recommendation about having more errors specific for some files.

Reply:
We can definitely accomplish low on the dependency tree _and_ just have lots of different leaves at the same time.